src/slice.cr

```diff
@@ -354,6 +354,26 @@ struct Slice(T)
     source.move_to(self)
   end
 
+  def gsub(buffer, &block : T -> _)
```

Comment:
This only compiles if `T` is `UInt8`. Are we sure we want this on slice instead of string?

Reply:
It looks like a similar problem as in #6389 where we have methods that make sense only for `Slice(UInt8)`, right?

Reply:
It's somewhat similar. But in contrast to #6389, this method conceptually fits for any slice. It's only that the `buffer` API is currently implemented by `IO` which is based on `UInt8`. But you can implement such a receiver for any type.

I'm not entirely sure about it's usefulnes, so maybe it's still better to move it to `String` as a private method.